packages/teams-js/test/internal/communication.spec.ts

```diff
@@ -1188,4 +1189,105 @@ describe('Testing communication', () => {
       });
     });
   });
+  describe('validateOrigin', () => {
+    it("validateOrigin returns false if the protocol of origin is not 'https:'", () => {
+      /* eslint-disable-next-line @microsoft/sdl/no-insecure-url */ /* Intentionally using http here because of what it is testing */
+      const messageOrigin = new URL('http://teams.microsoft.com');
+      const result = validateOrigin(messageOrigin);
+      expect(result).toBe(false);
+    });
+    it('validateOrigin returns true if origin is in teams pre-known allowlist', () => {
+      const messageOrigin = new URL('https://teams.microsoft.com');
+      const result = validateOrigin(messageOrigin);
+      expect(result).toBe(true);
+    });
+    it('validateOrigin returns true if origin for subdomains in teams pre-known allowlist', () => {
+      const messageOrigin = new URL('https://subdomain.teams.microsoft.com');
+      const result = validateOrigin(messageOrigin);
+      expect(result).toBe(true);
+    });
+    it('validateOrigin returns false if origin is not in teams pre-known allowlist', () => {
+      const messageOrigin = new URL('https://badorigin.com');
```

Comment:
>https://badorigin.com') [](http://example.com/codeflow?start=37&length=23)

Generally speaking we should only put origins that exist for the purposes of testing (e.g., [example.com](https://en.wikipedia.org/wiki/Example.com)) into our tests. I don't know what is at badorigin.com but we don't want people accidentally visiting places not under our control, even if they might be okay today.

It's probably okay to use microsoft.com addresses, but the example.* domains are intended to us for this purpose.